src/proc/threshold.cpp

```diff
@@ -0,0 +1,64 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+#include "../include/librealsense2/hpp/rs_sensor.hpp"
+#include "../include/librealsense2/hpp/rs_processing.hpp"
+
+#include "proc/synthetic-stream.h"
+#include "context.h"
+#include "environment.h"
+#include "option.h"
+#include "threshold.h"
+#include "image.h"
+
+namespace librealsense
+{
+    threshold::threshold() : _min(0.15f), _max(4.f)
+    {
+        auto min_opt = std::make_shared<ptr_option<float>>(0.f, 16.f, 0.1f, 0.15f, &_min, "Min range in meters");
+        register_option(RS2_OPTION_MIN_DISTANCE, min_opt);
+
+        auto max_opt = std::make_shared<ptr_option<float>>(0.f, 16.f, 0.1f, 4.f, &_max, "Max range in meters");
+        register_option(RS2_OPTION_MAX_DISTANCE, max_opt);
+    }
+
+    rs2::frame threshold::process_frame(const rs2::frame_source& source, const rs2::frame& f)
+    {
+        if (!f.is<rs2::depth_frame>()) return f;
```

Comment:
The processing is set for depth distances, but it should guard/handle depth in disparity domain.
Imho throwing "not implemented" is sufficient as it is rather improper usage of this filter.

Reply:
Added stream_filter to ctor to filter out all non depth frames